library/core/src/fmt/num.rs

```diff
@@ -199,6 +200,18 @@ static DEC_DIGITS_LUT: &[u8; 200] = b"\
       6061626364656667686970717273747576777879\
       8081828384858687888990919293949596979899";
 
+unsafe fn slice_buffer_to_str(buf: &[MaybeUninit<u8>], offset: usize) -> &str {
+    // SAFETY: All buf content since offset is set.
```

Comment:
this comment doesn't make sense here? this function probably needs to state its preconditions.